doctr/datasets/synthtext.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (C) 2021, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+import os
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import numpy as np
+import scipy.io as sio
+from tqdm import tqdm
+
+from .datasets import VisionDataset
+
+__all__ = ['SynthText']
+
+
+class SynthText(VisionDataset):
+    """SynthText dataset from `"Synthetic Data for Text Localisation in Natural Images"
+    <https://arxiv.org/abs/1604.06646>`_.
+
+    Example::
+        >>> from doctr.datasets import SynthText
+        >>> train_set = SynthText(train=True, download=True)
+        >>> img, target = train_set[0]
+
+    Args:
+        train: whether the subset should be the training one
+        sample_transforms: composable transformations that will be applied to each image
+        rotated_bbox: whether polygons should be considered as rotated bounding box (instead of straight ones)
+        **kwargs: keyword arguments from `VisionDataset`.
+    """
+
+    URL = 'https://thor.robots.ox.ac.uk/~vgg/data/scenetext/SynthText.zip'
+    #SHA256 = '28ab030485ec8df3ed612c568dd71fb2793b9afbfa3a9d9c6e792aef33265bf1'
+
+    def __init__(
+        self,
+        train: bool = True,
+        sample_transforms: Optional[Callable[[Any], Any]] = None,
+        rotated_bbox: bool = False,
+        **kwargs: Any,
+    ) -> None:
+
+        super().__init__(url=self.URL, file_name=None, file_hash=None, extract_archive=True, **kwargs)
+        self.sample_transforms = sample_transforms
+        self.train = train
+
+        # Load mat data
+        tmp_root = os.path.join(self.root, 'SynthText')
+        mat_data = sio.loadmat(os.path.join(tmp_root, 'gt.mat'))
+        split = int(len(mat_data['imnames'][0]) * 0.9)
+        paths = mat_data['imnames'][0][:split] if self.train else mat_data['imnames'][0][split:]
+        boxes = mat_data['wordBB'][0][:split] if self.train else mat_data['wordBB'][0][split:]
+        labels = mat_data['txt'][0][:split] if self.train else mat_data['txt'][0][split:]
+
+        self.data: List[Tuple[str, Dict[str, Any]]] = []
+        np_dtype = np.float32
+
+        for img_path, word_boxes, txt in tqdm(iterable=zip(paths, boxes, labels),
+                                              desc='Loading SynthText...', total=len(paths)):
+
+            # File existence check
+            if not os.path.exists(os.path.join(tmp_root, img_path[0])):
+                raise FileNotFoundError(f"unable to locate {os.path.join(tmp_root, img_path[0])}")
+
+            labels = '\n'.join(txt).split()
```

Comment:
why are we joining only to split?
Can't we split + strip directly?

Reply:
Mhh no `txt ` is a really ugly np.array :sweat_smile: 
example:
 array(['5FG      ', 'the      ', 'either   ', 'Sincerely', 'cars     ',
       'From:    ', 'the      ', 'and\nthat ', 'old      '], dtype='<U9')]

Reply:
I don't get how 1box label can have `\n` in it. Especially if we split it afterwards, 1 box = 2 text labels?
for your example, this would be more efficient:
```python
labels = [elt for word in txt.tolist() for elt in word.split()]
```

Reply:
applied